lib/engine/game/g_18_ireland/game.rb

```diff
@@ -394,6 +394,20 @@ def upgrade_cost(old_tile, hex, entity)
           super
         end
 
+        def unstarted_corporation_summary
+          unipoed = @corporations.reject(&:ipoed)
+          minor = unipoed.select { |c| c.type == :minor }
+          major = unipoed.select { |c| c.type == :major }
+          ["#{major.size} major", minor]
+        end
+
+        def timeline
+          timeline = []
+          minors = @corporations.reject(&:ipoed).select { |c| c.type == :minor }.map(&:name)
```

Comment:
Check both conditions in the select, so you only have to iterate through the array once.